Redesigned UI

Vimall03 opened the pull request:
This pull request introduces a complete redesign of the landing page, login page, and signup page. The redesign focuses on improving responsiveness and user experience across various devices.

### Changes Made

1. Replaced existing landing page with a redesigned version that enhances visual appeal and usability.
2. Revamped login page to streamline the login process and improve accessibility.
3. Implemented a redesigned signup page with clear instructions and intuitive form fields.
4. Ensured responsiveness across desktop, tablet, and mobile devices.

Closes #70

abhijeetnishal commented on client/public/assets/Illustration-1-dark.svg:
Create a separate folder inside components named icons and inside this create a .tsx file and use this svg as an component like:

```tsx

export const IllustrationIcon = ()=>{
  return (
    <svg width="60" height="60" viewBox="0 0 60 60" fill="none" xmlns="http://www.w3.org/2000/svg">
<rect width="60" height="60" rx="3" fill="#FABB18"/>
<path d="M29.7143 15C25.5357 15 22.2143 18.3214 22.2143 22.5V23.5714C20.3929 23.5714 19 24.9643 19 26.7857V34.2857C19 40.1786 23.8214 45 29.7143 45C35.6071 45 40.4286 40.1786 40.4286 34.2857V26.7857C40.4286 24.9643 39.0357 23.5714 37.2143 23.5714V22.5C37.2143 18.3214 33.8929 15 29.7143 15ZM38.2857 26.7857V34.2857C38.2857 39 34.4286 42.8571 29.7143 42.8571C25 42.8571 21.1429 39 21.1429 34.2857V26.7857C21.1429 26.1429 21.5714 25.7143 22.2143 25.7143H23.2857H36.1429H37.2143C37.8571 25.7143 38.2857 26.1429 38.2857 26.7857ZM24.3571 23.5714V22.5C24.3571 19.5 26.7143 17.1429 29.7143 17.1429C32.7143 17.1429 35.0714 19.5 35.0714 22.5V23.5714H24.3571Z" fill="white"/>
<path d="M28.8577 31.8217H32.3934C33.0363 31.8217 33.4648 31.3931 33.4648 30.7503C33.4648 30.1074 33.0363 29.6789 32.3934 29.6789H30.7863V28.9289C30.7863 28.286 30.3577 27.8574 29.7148 27.8574C29.072 27.8574 28.6434 28.286 28.6434 28.9289V29.6789C27.1434 29.786 25.9648 30.9646 25.9648 32.4646C25.9648 34.0717 27.2506 35.3574 28.8577 35.3574H30.6791C31.1077 35.3574 31.4291 35.6789 31.4291 36.1074C31.4291 36.536 31.1077 36.8574 30.6791 36.8574H27.1434C26.5006 36.8574 26.072 37.286 26.072 37.9289C26.072 38.5717 26.5006 39.0003 27.1434 39.0003H28.6434V39.6431C28.6434 40.286 29.072 40.7146 29.7148 40.7146C30.3577 40.7146 30.7863 40.286 30.7863 39.6431V38.8931C32.2863 38.786 33.4648 37.6074 33.4648 36.1074C33.4648 34.5003 32.1791 33.2146 30.572 33.2146H28.7506C28.322 33.2146 28.0006 32.8931 28.0006 32.4646C28.1077 32.1431 28.4291 31.8217 28.8577 31.8217Z" fill="white"/>
</svg>
  )
}
```

abhijeetnishal commented on client/public/assets/Illustration-2.svg:
Same for all svg's

abhijeetnishal commented on client/public/assets/Illustration-3.svg:
I think this illustration is not fit for our project. Use or replace with other illustration.

abhijeetnishal commented on client/src/app/layout.tsx:
```diff
@@ -1,24 +1,67 @@
+}
+
+// Create the context with a default value
+const ThemeContext = createContext<ThemeContextType | undefined>(undefined);
```
Use `**zustand**` for state management not context api. Reason zustand is so lightweight and efficient and give better rendering with less bundelling size. Currently our project is using zustand got to the folder named stores you can see. 

Also create this theme as themeStore in stores folder instead of layout. Reason layout should be Server component.

abhijeetnishal commented on client/src/app/layout.tsx:
```diff
@@ -1,24 +1,67 @@
+}
+
+// Create a provider component
+const ThemeProvider: FC<ThemeProviderProps> = ({ children }) => {
```
As discussed theme should in stores folder

abhijeetnishal commented on client/src/components/Headers/Navbar.tsx:
```diff
@@ -0,0 +1,99 @@
+              <button className='p-1 md:mt-0 mt-2' onClick={toggleTheme}>
+                {
+                  theme === 'light' && 
+                  <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24"><path d="M12 10.999c1.437.438 2.562 1.564 2.999 3.001.44-1.437 1.565-2.562 3.001-3-1.436-.439-2.561-1.563-3.001-3-.437 1.436-1.562 2.561-2.999 2.999zm8.001.001c.958.293 1.707 1.042 2 2.001.291-.959 1.042-1.709 1.999-2.001-.957-.292-1.707-1.042-2-2-.293.958-1.042 1.708-1.999 2zm-1-9c-.437 1.437-1.5 ...
```
As discussed svg's should be in icons folder and used as a component. Reason to maintain folder structure.

abhijeetnishal commented on client/src/components/Headers/Navbar.tsx:
```diff
@@ -0,0 +1,99 @@
+            <li>
+              <button className='p-1 md:mt-0 mt-2' onClick={toggleTheme}>
+                {
+                  theme === 'light' && 
```
use ternary operator like this everywhere: 
`theme === 'light' ? <svg> : null` 

Reason it's a good practice to use this.

abhijeetnishal commented on client/src/components/Footer/Footer.tsx:
```diff
@@ -0,0 +1,43 @@
+"use client";
```
Remove this "use client"

Footer in server components it's not necessary always to use "use client" only use in client component.

abhijeetnishal commented on client/src/app/auth/login/page.tsx:
```diff
@@ -64,63 +72,76 @@ const Login = () => {
     }
   };
-
+  const { theme, toggleTheme } = useThemeContext();
```
As discussed use zustand stores approach everywhere as discussed.

abhijeetnishal commented on client/src/app/auth/login/page.tsx:
```diff
@@ -64,63 +72,76 @@ const Login = () => {
+          </form>
+          <div className="py-12 text-center">
+            <p className="text-gray-600">
+              Don't have an account?
```
react not support this:  `don't` use like this `don&apos;t` 
Reason it will fails the build. 
Fix this where ever we are using this '

abhijeetnishal commented:
It's not responsive in tablet. make little responsive if possible.

![Screenshot 2024-05-29 205011](https://github.com/abhijeetnishal/Password-Shield/assets/83833440/bea8eb3e-2801-4818-b53a-f267ac6c1a35)

Vimall03 commented:
@abhijeetnishal 

Sure, I'll implement the changes and look forward to learning new things.

Vimall03 commented:
> > @abhijeetnishal
> > Sure, I'll implement the changes and look forward to learning new things.
> 
> That great @Vimall03, you are taking this as a learning not as a just completing as most of the contributors does that.
> 
> If you need any help let me know.

Thank you, @abhijeetnishal. I have made the mentioned changes. Let me know if it is good.

abhijeetnishal commented:
> > > @abhijeetnishal
> > > Sure, I'll implement the changes and look forward to learning new things.
> > 
> > 
> > That great @Vimall03, you are taking this as a learning not as a just completing as most of the contributors does that.
> > If you need any help let me know.
> 
> Thank you, @abhijeetnishal. I have made the mentioned changes. Let me know if it is good.

Hey @Vimall03, I will review the code in the evening currently I am at work.

abhijeetnishal commented:
Hey @Vimall03, the code level changes i mentioned you have done nicely, i appreciate it.

There are some responsive changes are required:
For checking responsiveness use this extension: https://chromewebstore.google.com/detail/mobile-simulator-responsi/ckejmhbmlajgoklhgbapkiccekfoccmk

1. Remove padding or margin to take full width for text.
![Screenshot 2024-05-30 202021](https://github.com/abhijeetnishal/Password-Shield/assets/83833440/9728c304-8b86-436e-8655-baa440bb8cf4)

2. Make this container flex col at this viewport i thing it's taking flex row. 
![Screenshot 2024-05-30 202034](https://github.com/abhijeetnishal/Password-Shield/assets/83833440/cf88cc66-2f12-48eb-b2df-b0b76c576c10)

abhijeetnishal commented:
Hey @Vimall03, Once done, let me know i will merge this PR.

Vimall03 commented:
Thank you for the extensions. Sure, I'll let you know once it's done.

Vimall03 commented:
Hey, @abhijeetnishal,

The extensions you suggested were really helpful. Thank you! I have fixed the layout issues—fingers crossed. I have also added the screenshots of the updated layout.

![image](https://github.com/abhijeetnishal/Password-Shield/assets/109758195/de10f2b8-0421-490f-acfb-7068c1cbcd9a)
![image](https://github.com/abhijeetnishal/Password-Shield/assets/109758195/49334bc0-62d2-46f2-a155-d705b40a74cf)

abhijeetnishal commented:
Hey @Vimall03, I will check and merge if anything is needs to be done I will do don't worry as you have done great job. 

Hey, i want you to contribute more in this project after seeing your impression work, as you have done the initial redesigning the inside pages also needs to be redesigned as it should match the design.
 I will make this issue as level3, take your time and let me know, it's your choice, I am not constraining you to work. You will be my valuable contributer always.

Vimall03 commented:
@abhijeetnishal 

I would definitely love to contribute more. I really enjoyed working on this project and appreciate the opportunity to continue. Thank you for considering me for this important task.

abhijeetnishal commented:
Great @Vimall03. Go through the projects and accordingly share the designs, we will verify and implement.

abhijeetnishal commented:
I am changing levels now for both PR and issue

Vimall03 commented:
> Great @Vimall03. Go through the projects and accordingly share the designs, we will verify and implement.

Sure

abhijeetnishal commented:
Hey @Vimall03, I have added some changes please pull the latest code so that the conflicts also get resolved.

abhijeetnishal commented:
Hey @Vimall03, try to resolve the conflicts, never try to force push the changes it will force update with the previous code sometimes. So try to learn resolving the conflicts. This is how open source collaborations work.

Vimall03 commented:
ok I'll keep that in mind. Also I will update you with the design for the dashboard by tomorrow.

abhijeetnishal commented:
> ok I'll keep that in mind. Also I will update you with the design for the dashboard by tomorrow.

That's Great.

Vimall03 commented:
Hey, @abhijeetnishal 
I used the postgressql server from render in my instance. But I am unable to figure out this error that I am getting
I am able to normally post and get requests through thunderbird but the client server keeps giving me this error

Can you please fix this ?

![image](https://github.com/abhijeetnishal/Password-Shield/assets/109758195/2860330a-5a2d-482c-9191-b5736ceed907)

ThunderClient response:
![Screenshot 2024-06-02 192756](https://github.com/abhijeetnishal/Password-Shield/assets/109758195/5da18a83-4bc4-49e5-bbee-fccfba099e31)

abhijeetnishal commented:
Hey @Vimall03, if it is working for postman like client it means the database issue is resolved. If you are facing this issue on client I think the issue is because of base url, have you added .env local file in nextjs app and added the env from .env sample ?

Vimall03 commented:
Yeah I have added the .env file 
And made no changes to the base url nor the port of the server

Vimall03 commented:
Also take a look at the dashboard design in the figma file. 
I'll implement the design for light mode directly in development.

If possible can you merge this PR for landing page and I
will open a new PR for the dashboard. You can assign a the levels accordingly.

https://www.figma.com/design/T5ZB3emjxim4JMAqnGQIsZ/GSSoC-KeySafe?node-id=0-1&t=oKrqm9g2lLbQnBlI-1

abhijeetnishal commented:
> Yeah I have added the .env file And made no changes to the base url nor the port of the server

So you are created .env.local file right not .env in client(next.js) ?

It should like this:
![Screenshot 2024-06-02 201744](https://github.com/abhijeetnishal/Password-Shield/assets/83833440/c213284f-690e-433f-8f63-945eda53393a)

NEXT_PUBLIC_BASE_URL=http://localhost:8080

abhijeetnishal commented:
> Also take a look at the dashboard design in the figma file. I'll implement the design for light mode directly in development.
> 
> If possible can you merge this PR for landing page and I will open a new PR for the dashboard. You can assign the levels accordingly.
> 
> https://www.figma.com/design/T5ZB3emjxim4JMAqnGQIsZ/GSSoC-KeySafe?node-id=0-1&t=oKrqm9g2lLbQnBlI-1

Design related things we can discuss on the issue also. In PR we can discuss code related stuffs 😅

abhijeetnishal commented:
> Also take a look at the dashboard design in the figma file. I'll implement the design for light mode directly in development.
> 
> If possible can you merge this PR for landing page and I will open a new PR for the dashboard. You can assign the levels accordingly.
> 
> https://www.figma.com/design/T5ZB3emjxim4JMAqnGQIsZ/GSSoC-KeySafe?node-id=0-1&t=oKrqm9g2lLbQnBlI-1

The reason i am not merging the landing page is if someone login then we can't show the old design it's a bad user experience. So after dashboard we can merge both. Don't worry i changed the label to level3 which is impactful.

Vimall03 commented:
> > Yeah I have added the .env file And made no changes to the base url nor the port of the server
> 
> So you are created .env.local file right not .env in client(next.js) ?
> 
> It should like this: ![Screenshot 2024-06-02 201744](https://private-user-images.githubusercontent.com/83833440/335874957-c213284f-690e-433f-8f63-945eda53393a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTczNDAwMDAsIm5iZiI6MTcxNzMzOTcwMCwicGF0aCI6Ii84MzgzMzQ0MC8zMzU4NzQ5NTctYzIxMzI4NGYtNjkwZS00MzNmLThmNjMtOTQ1ZWRhNTMzOTNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjAyVDE0NDgyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE2MDU4OTRkYjM3YWVhNWJiM2M2ZWNmNDI2MjRkYjczZTIxNjM2MGEyNjk0Y2ZiOGE5ODQxMTAwNzA2M2U1NGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Suq_KDV9OwZLu-jFPiLRUcp5kggfUZuq1bZIOLbgozo)
> 
> NEXT_PUBLIC_BASE_URL=http://localhost:8080

I tried adding the .env.local file as well

Im getting the same error.
![image](https://github.com/abhijeetnishal/Password-Shield/assets/109758195/2dbfe115-0686-4950-9e37-7bccaafa7b28)

![image](https://github.com/abhijeetnishal/Password-Shield/assets/109758195/b4a53254-b2d5-4eea-adae-fd55d90c015b)

abhijeetnishal commented:
Hey @Vimall03, do one thing clone this repo and try to run the client if it works then it's fine means there will be issue in forked repo code. 
Let me once done.

Vimall03 commented:
Okay i will do that
